platform/core/PolyPodCoreSwift/Sources/PolyPodCoreSwift/MessagePack+Decoding.swift

```diff
@@ -0,0 +1,65 @@
+import MessagePack
+
+extension MessagePackValue {
+    func getDictionary() throws -> [MessagePackValue: MessagePackValue]? {
+        guard self != nil else {
+            return nil
+        }
+        
+        if let dictionary = self.dictionaryValue {
+            return dictionary
+        } else {
+            throw DecodingError.invalidValue(info: "Expected dictionary")
+        }
+    }
+    
+    func getString() throws -> String? {
+        guard self != nil else {
+            return nil
+        }
+        if let string = self.stringValue {
+            return string
+        } else {
+            throw DecodingError.invalidValue(info: "Expected string")
+        }
+    }
+    
+    func getString() throws -> String {
+        if let string = self.stringValue {
+            return string
+        } else {
+            throw DecodingError.invalidValue(info: "Expected string")
+        }
+    }
+}
+
+func mapError(_ dict: [MessagePackValue: MessagePackValue]) throws -> CoreFailure {
+    guard let code = dict["code"]?.intValue.flatMap(CoreFailureCode.init),
+          let message = dict["message"]?.stringValue else {
+        throw DecodingError.invalidCoreFailure(info: "Received \(dict)")
+    }
+    
+    return CoreFailure(code: code, message: message)
+}
+
+func mapFeatureManifest(_ value: MessagePackValue) throws -> FeatureManifest {
```

Comment:
We should be able to leverage `Codable` Swift protocol to automatically Encode/Decode Swift models.